src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -242,33 +257,61 @@ async Task<JObject> ResolveAsLocalOrThisMember(string name)
                     return null;
                 }
 
-                JToken objRet = valueOrError.Value.FirstOrDefault(objPropAttr => objPropAttr["name"].Value<string>() == nameTrimmed);
+                JToken objRet = valueOrError.Value.FirstOrDefault(objPropAttr => objPropAttr["name"].Value<string>() == name);
                 if (objRet != null)
                     return await GetValueFromObject(objRet, token);
 
                 return null;
             }
 
-            async Task<JObject> ResolveAsInstanceMember(string expr, JObject baseObject)
+            async Task<JObject> ResolveAsInstanceMember(ArraySegment<string> parts, JObject baseObject, bool hasNullCondition)
             {
+                // if the prevPart has "?" at the end: true
+                bool hasResolvedObjNullCondition = hasNullCondition;
+                bool isPrevPartNull = false;
                 JObject resolvedObject = baseObject;
-                string[] parts = expr.Split('.');
-                for (int i = 0; i < parts.Length; i++)
+                for (int i = 0; i < parts.Count; i++)
                 {
-                    string partTrimmed = parts[i].Trim();
+                    string partTrimmed = parts[i];
                     if (partTrimmed.Length == 0)
                         return null;
 
+                    // current value of resolvedObject is on parts[i - 1]
+                    if (isPrevPartNull || resolvedObject.IsNullValuedObject())
+                    {
+                        // trying null.$member
+                        if (!hasResolvedObjNullCondition)
+                            throw new ReturnAsErrorException($"Cannot access member \"{partTrimmed}\" of a null-valued object.", "ReferenceError");
```

Comment:
Error can be more explicit about what happened - `"Expression threw NullReferenceException trying to access \"{partTrimmed}\" on a null-valued object.`.